lib/compiler.js

```diff
@@ -499,14 +517,17 @@ function buildMiddlewareInstructions(rootDir, config) {
   };
 }
 
-function resolveMiddlewarePath(rootDir, middleware) {
+function resolveMiddlewarePath(rootDir, middleware, config) {
   var resolved = {};
 
   var segments = middleware.split('#');
   var pathName = segments[0];
   var fragment = segments[1];
   var middlewarePath = pathName;
-  var opts = { strict: true };
+  var opts = {
+    strict: true,
+    optional: config.optional
```

Comment:
I used the value assuming, it might be needed somewhere down the line. Value coercion should suffice in the current implementation. Will change it.